src/cpp/benders/benders_mpi/BendersMPI.cpp

```diff
@@ -131,19 +136,49 @@ void BendersMpi::step_2_solve_subproblems_and_build_cuts() {
 
 void BendersMpi::gather_subproblems_cut_package_and_build_cuts(
     const SubProblemDataMap &subproblem_data_map, const Timer &walltime) {
-  if (!_exceptionRaised) {
+  if (!exception_raised_) {
     std::vector<SubProblemDataMap> gathered_subproblem_map;
     mpi::gather(_world, subproblem_data_map, gathered_subproblem_map, rank_0);
     SetSubproblemsWalltime(walltime.elapsed());
     double cumulative_subproblems_timer_per_iter(0);
     Reduce(GetSubproblemsCpuTime(), cumulative_subproblems_timer_per_iter,
            std::plus<double>(), rank_0);
     SetSubproblemsCumulativeCpuTime(cumulative_subproblems_timer_per_iter);
+    if (Options().EXTERNAL_LOOP_OPTIONS.DO_OUTER_LOOP) {
+      _data.outer_loop_current_iteration_data.outer_loop_criterion =
+          ComputeSubproblemsContributionToOuterLoopCriterion(
+              subproblem_data_map);
+      if (_world.rank() == rank_0) {
+        outer_loop_criterion_.push_back(
+            _data.outer_loop_current_iteration_data.outer_loop_criterion);
+        UpdateOuterLoopMaxCriterionArea();
+      }
+    }
     // only rank_0 receive non-emtpy gathered_subproblem_map
     master_build_cuts(gathered_subproblem_map);
   }
 }
 
+std::vector<double>
+BendersMpi::ComputeSubproblemsContributionToOuterLoopCriterion(
```

Comment:
I would invert the names `ComputeOuterLoopContribution` and `ComputeSubproblemsContributionToOuterLoopCriterion` as for now the former acts only on single subproblems whereas the latter performs reduction over all subproblems